index.php

```diff
@@ -4,73 +4,121 @@
 # SPDX-License-Identifier: AGPL-3.0
 #
 
+// function to make http GET requests
+function makeRequest($username, $token, $url)
+{
+    // init curl
+    $ch = curl_init($url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+
+    // set headers
+    $headers = array("Authorization: $username:$token");
+    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
```

Comment:
This will break if there is a `:` char inside username or token.
The request should be made using Base64. Something like:
```
curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
```